string.go

```diff
@@ -30,40 +28,45 @@ import (
 	"unicode/utf8"
 )
 
-// AESEncrypt encrypts text and given key with AES.
-func AESEncrypt(key, text []byte) ([]byte, error) {
+// AESEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
+func AESEncrypt(key, nonce, plaintext []byte) ([]byte, error) {
```

Comment:
How can the nonce be automatically generated and encapsulated in here? It needs to be unique for every encryption and be passed over for decryption purposes. I guess there could be methods in here to generate and fetch the nonce, but the client would still need to call them before using encrypt/decrypt. Another option would be to change the structure of this code was so you got an instance. Then you could make use of a counter that's incremented every time encrypt is called.